### PR TITLE
Csv generator

### PR DESCRIPTION
**Pull request by enricoliscio**
Added a new Generator specific for masks, and a new CSVGenerator on top of it. 
@hgaiser

### CODE REVIEW

File: keras_maskrcnn/bin/test_csv_generator.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from keras_maskrcnn.preprocessing.csv_generator import CSVGenerator
```
Comment (hgaiser) [inline keras_maskrcnn/bin/test_csv_generator.py]: What does this tool do that `debug.py` doesn't do?

File: keras_maskrcnn/preprocessing/csv_generator.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+            with _open_for_csv(csv_class_file) as file:
+                self.classes = _read_classes(csv.reader(file, delimiter=','))
+        except ValueError as e:
+            raise_from(ValueError('invalid CSV class file: {}: {}'.format(csv_class_file, e)), None)
+
+        self.labels = {}
+        for key, value in self.classes.items():
+            self.labels[value] = key
+
+        # csv with img_path, x1, y1, x2, y2, class_name
```
Comment (hgaiser) [inline keras_maskrcnn/preprocessing/csv_generator.py]: Not correct anymore.

File: keras_maskrcnn/preprocessing/csv_generator.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        masks        = []
+
+        for idx, annot in enumerate(annots):
+            annotations[idx, 0]  = float(annot['x1'])
+            annotations[idx, 1]  = float(annot['y1'])
+            annotations[idx, 2]  = float(annot['x2'])
+            annotations[idx, 3]  = float(annot['y2'])
+            annotations[idx, 4]  = self.name_to_label(annot['class'])
+            mask = cv2.imread(annot['mask_path'], cv2.IMREAD_GRAYSCALE).astype(float)
+            mask[np.where(mask > 0)] = 1  # convert from 0-255 to binary mask
```
Comment (hgaiser) [inline keras_maskrcnn/preprocessing/csv_generator.py]: This can be simplified to `mask = (mask > 0).astype(np.uint8)`.

File: keras_maskrcnn/preprocessing/generator.py [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+        regression_batch = np.zeros((self.batch_size,) + regression_group[0].shape, dtype=keras.backend.floatx())
+
+        # copy all labels and regression values to the batch blob
+        for index, (labels, regression) in enumerate(zip(labels_group, regression_group)):
+            labels_batch[index, ...]     = labels
+            regression_batch[index, ...] = regression
+
+        # copy all annotations / masks to the batch
+        max_annotations = max(a.shape[0] for a in annotations_group)
+        masks_batch     = np.zeros((self.batch_size, max_annotations, 5 + 2 + max_shape[0] * max_shape[1]), dtype=keras.backend.floatx())
```
Comment (hgaiser) [inline keras_maskrcnn/preprocessing/generator.py]: Maybe document somewhere what this strange shape means.

File: keras_maskrcnn/bin/test_csv_generator.py (hunk 1 above)
Comment (enricoliscio) [reply keras_maskrcnn/bin/test_csv_generator.py]: I just tested a couple of things, it was sort of a unit testing. I can checkthe other one and add a proper unit testing.

File: keras_maskrcnn/preprocessing/csv_generator.py (hunk 2 above)
Comment (enricoliscio) [reply keras_maskrcnn/preprocessing/csv_generator.py]: well, it was 6 days ago when I sent the pull request

Comment (hgaiser) [reply keras_maskrcnn/preprocessing/csv_generator.py]: Hmm but six days ago this also included the mask_path?

Comment (enricoliscio) [reply keras_maskrcnn/preprocessing/csv_generator.py]: wait, but are you talking about the comment?

Comment (hgaiser) [reply keras_maskrcnn/preprocessing/csv_generator.py]: yeah

File: keras_maskrcnn/preprocessing/csv_generator.py (hunk 3 above)
Comment (enricoliscio) [reply keras_maskrcnn/preprocessing/csv_generator.py]: done.

File: keras_maskrcnn/preprocessing/generator.py (hunk 4 above)
Comment (enricoliscio) [reply keras_maskrcnn/preprocessing/generator.py]: done.

File: keras_maskrcnn/bin/test_csv_generator.py (hunk 1 above)
Comment (hgaiser) [reply keras_maskrcnn/bin/test_csv_generator.py]: Yeah I think it's better to remove this and add actual unit testing.
